apps/prairielearn/src/models/course-instances.ts

```diff
@@ -96,3 +97,21 @@ export async function selectCourseInstanceGraderStaff({
     minimal_role: 'Student Data Editor',
   });
 }
+
+/**
+ * Returns if the course has any non-deleted course instances.
+ */
+export async function selectCourseHasCourseInstances({
+  course_id,
+}: {
+  course_id: string;
+}): Promise<boolean> {
+  // Check if the course has any non-deleted course instances.
+  const hasCourseInstances = await queryRow(
+    sql.select_course_has_course_instances,
+    { course_id },
+    z.boolean(),
+  );
+
+  return hasCourseInstances;
```

Comment:
Nits:

- Comment is somewhat redundant with the JSDoc comment on the function as a whole.
- No need for an intermediate local variable.

```suggestion
  return await queryRow(
    sql.select_course_has_course_instances,
    { course_id },
    z.boolean(),
  );
```